domain-management/src/main/java/org/jboss/as/domain/management/access/ApplicationClassificationConfigResourceDefinition.java

```diff
@@ -121,17 +119,29 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
 
         @Override
         public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
-            final String attribute = operation.require(NAME).asString();
+            PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
+            ModelNode modelNode = context.readResourceFromRoot(address).getModel();
+            // record model values for rollback handler
+            ModelNode configuredApplication = modelNode.get(ModelDescriptionConstants.CONFIGURED_APPLICATION);
```

Comment:
These lines can come after L128 and the first line becomes

ModelNode modelNode = resource.getModel();

The Resource from context.readResourceForUpdate includes the data that you need to restore.